packages/editor/src/components/post-actions/actions.js

```diff
@@ -64,23 +110,36 @@ const trashPostAction = {
 		const { createSuccessNotice, createErrorNotice } =
 			useDispatch( noticesStore );
 		const { deleteEntityRecord } = useDispatch( coreStore );
+		const { __experimentalDeleteReusableBlock } =
+			useDispatch( reusableBlocksStore );
+		const isPattern = items[ 0 ].type === PATTERN_POST_TYPE;
+		const deleteItem = isPattern
+			? ( item ) => __experimentalDeleteReusableBlock( item.id )
```

Comment:
This action removes the reusable block but also updates the current block editor to remove that block. It feels a bit misplaced for me. A delete reusable block action should just delete the reusable block and not impact the block editor IMO. So maybe we should just unify the calls here.

Reply:
You mean using only the delete entity call, right? I didn't investigate why it was introduced, but it seemed it has some specific logic that might be still needed. I'll check it more.